src/ucp/core/ucp_mm.c

```diff
@@ -150,9 +151,8 @@ static ucs_status_t ucp_mem_alloc(ucp_context_h context, size_t length,
             }
         }
 
-        /* TODO: pass addr */
-        status = uct_mem_alloc(NULL, length, uct_flags, &method, 1, mds, num_mds,
-                               name, &mem);
+        status = uct_mem_alloc(memh->address, length, uct_flags, &method, 1, mds,
```

Comment:
@shamisp the API doc says "use address as a hint". If the memory is allocated in a different address - user can find this using ucp_mem_query(), it is not considered an error situation.